## Assignment 4

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv(r"C:\Users\HP i5\Desktop\SE_Factory\Week 1\Pythonex\kc_house_data.csv")

In [4]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [5]:
df['date'] = pd.to_datetime(df['date'])

To start with the analysis, i began with comparing the attributes and features of the most expensive houses with those of the cheapest houses. in order to get some initial insights on what to focus on through the analysis of this dataset.

In [6]:
df.nlargest(5, 'price')

id       date      price  bedrooms  bathrooms  sqft_living  \
7252  6762700020 2014-10-13  7700000.0         6       8.00        12050   
3914  9808700762 2014-06-11  7060000.0         5       4.50        10040   
9254  9208900037 2014-09-19  6890000.0         6       7.75         9890   
4411  2470100110 2014-08-04  5570000.0         5       5.75         9200   
1448  8907500070 2015-04-13  5350000.0         5       5.00         8000   

      sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
7252     27600     2.5           0     3  ...     13        8570   
3914     37325     2.0           1     2  ...     11        7680   
9254     31374     2.0           0     4  ...     13        8860   
4411     35069     2.0           0     0  ...     13        6200   
1448     23985     2.0           0     4  ...     12        6720   

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
7252           3480      1910          1987    98102  47.6298 -122.323   
3914           2360      1940          2001    98004  47.6500 -122.214   
9254           1030      2001             0    98039  47.6305 -122.240   
4411           3000      2001             0    98039  47.6289 -122.233   
1448           1280      2009             0    98004  47.6232 -122.220   

      sqft_living15  sqft_lot15  
7252           3940        8800  
3914           3930       25449  
9254           4540       42730  
4411           3560       24345  
1448           4600       21750  

[5 rows x 21 columns]

The subsets with the highest prices (Above 7 million dollars) have various number of bedrooms/bathrooms, high living area, good view levels, higher grades (above 11) significantly higher basement area. All of these may contribute to the pricing of these properties, which is often true as more expensive houses tend to have luxurious features.  

In [7]:
df.nsmallest(5, 'price')

id       date    price  bedrooms  bathrooms  sqft_living  \
1149   3421079032 2015-02-17  75000.0         1       0.00          670   
15293    40000362 2014-05-06  78000.0         2       1.00          780   
465    8658300340 2014-05-23  80000.0         1       0.75          430   
16198  3028200080 2015-03-24  81000.0         2       1.00          730   
8274   3883800011 2014-11-05  82000.0         3       1.00          860   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
1149      43377     1.0           0     0  ...      3         670   
15293     16344     1.0           0     0  ...      5         780   
465        5050     1.0           0     0  ...      4         430   
16198      9975     1.0           0     0  ...      5         730   
8274      10426     1.0           0     0  ...      6         860   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
1149               0      1966             0    98022  47.2638 -121.906   
15293              0      1942             0    98168  47.4739 -122.280   
465                0      1912             0    98014  47.6499 -121.909   
16198              0      1943             0    98168  47.4808 -122.315   
8274               0      1954             0    98146  47.4987 -122.341   

       sqft_living15  sqft_lot15  
1149            1160       42882  
15293           1700       10387  
465             1200        7500  
16198            860        9000  
8274            1140       11250  

[5 rows x 21 columns]

The subsets with the cheapest houses(Around 80,000$) have the very low area (less than one thousand) smaller number of bedrooms and bathrooms, only one floor, 0 waterfront, no view, low grade leveks (less than 7), and no basement. Those features may be influecing the pricing of the properties.  

After doing an initial analysis on the prices, i will apply the correlation function to get some additional insights on the relationship between each attribute and the target(price). The correlation function gives a values between 1(Strong positive correlation) to -1(Strong negative correlation) with 0 correlation meaning no relationship. It reflects how similar the measurements of variables are across a dataset and identifies patterns and relationships that exist within the data. However, the default .corr() function (which uses pearson's method) is usually applied on numeric variables.  

In [8]:
correlation = df.corr()
price_corr = correlation['price'].sort_values(ascending=False)
print(price_corr)

price            1.000000
sqft_living      0.702044
grade            0.667463
sqft_above       0.605566
sqft_living15    0.585374
bathrooms        0.525134
view             0.397346
sqft_basement    0.323837
bedrooms         0.308338
lat              0.306919
waterfront       0.266331
floors           0.256786
yr_renovated     0.126442
sqft_lot         0.089655
sqft_lot15       0.082456
yr_built         0.053982
condition        0.036392
long             0.021571
date            -0.004366
id              -0.016797
zipcode         -0.053168
Name: price, dtype: float64


The correlation method provides us with some good insights; there is a strong correlation between the price and the square foot of living space. However concerning the different variable we have; categorical variables and geographical variables, there isn't much to inspect through this method and it doesnt provide good values on such correlation. The df.corr() method works on continuous numerical variables, but not for ranked/geographic variables. 

In [9]:
correlation = df.corr(method='kendall')
price_corr = correlation['price'].sort_values(ascending=False)
print(price_corr)

price            1.000000
grade            0.527844
sqft_living      0.463620
sqft_living15    0.405037
sqft_above       0.378244
bathrooms        0.369216
lat              0.292903
bedrooms         0.265100
floors           0.251630
view             0.237574
sqft_basement    0.190399
waterfront       0.094071
yr_renovated     0.082378
yr_built         0.070321
sqft_lot         0.047191
long             0.039866
sqft_lot15       0.038425
condition        0.014485
id               0.003218
date            -0.007266
zipcode         -0.009826
Name: price, dtype: float64


This correlation shows that the categorical variable; bathrooms has a moderate correlation with the price, however the bedrooms, floors, and view show much weaker positive correlation implying that they may influence the price but to a lesser extent.  

In [10]:
df[['price', 'bedrooms', 'bathrooms', 'sqft_living']].groupby(['bedrooms', 'bathrooms']).mean().round().reset_index()



bedrooms  bathrooms      price  sqft_living
0           0       0.00   521707.0       2058.0
1           0       0.75   265000.0        384.0
2           0       1.00   228000.0        390.0
3           0       1.50   288000.0       1430.0
4           0       2.50   299983.0       1863.0
..        ...        ...        ...          ...
139        10       2.00   650000.0       3610.0
140        10       3.00   660000.0       2920.0
141        10       5.25  1150000.0       4590.0
142        11       3.00   520000.0       3000.0
143        33       1.75   640000.0       1620.0

[144 rows x 4 columns]

This dataframe shows that houses with more bedrooms and bathrooms are usually of higher area and higher prices. As shown in the table, some of the houses have zero bedrooms but different number of bathrooms, meaning that they are either a studio house or they is a mistake in the data implemented. Also there is a house with 33 bedrooms however a rather small area, so this must be a typo. 

In [11]:
df.loc[(df['bedrooms'] == 8)  ]

id       date      price  bedrooms  bathrooms  sqft_living  \
1660   9126101740 2014-12-04   490000.0         8       5.00         2800   
4035    685000115 2014-10-07  2150000.0         8       6.00         4340   
4067   7226500100 2015-02-19   373000.0         8       3.00         2850   
6174   1873400020 2014-07-03   340000.0         8       2.75         2790   
9077   6746700615 2015-03-18   700000.0         8       2.50         2280   
9452   2769600590 2014-10-16   900000.0         8       4.00         4020   
10958  1652500060 2014-07-11  1650000.0         8       2.75         4040   
12885  8813400155 2014-12-19   808000.0         8       3.75         3460   
15070  7418000130 2014-12-11   430000.0         8       3.25         4300   
15670  8141200080 2014-08-14   680000.0         8       2.75         2530   
17235  1338600225 2014-05-28  1970000.0         8       3.50         4440   
18477  1346300150 2014-10-20  3300000.0         8       4.00         7710   
19302  3756900027 2014-11-25   575000.0         8       3.00         3840   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
1660       2580     2.0           0     0  ...      8        1880   
4035       9415     2.0           0     0  ...      8        4340   
4067      12714     1.0           0     0  ...      7        2850   
6174       6695     1.0           0     0  ...      7        1470   
9077       3000     1.5           0     0  ...      7        1210   
9452       7500     1.0           0     0  ...      8        2010   
10958     20666     1.0           0     0  ...      9        2020   
12885      4600     2.0           0     0  ...      7        2860   
15070     10441     2.0           0     0  ...      8        2800   
15670      4800     2.0           0     0  ...      7        1390   
17235      6480     2.0           0     3  ...     10        3140   
18477     11750     3.5           0     0  ...     12        6090   
19302     15990     1.0           0     0  ...      7        2530   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
1660             920      1997             0    98122  47.6086 -122.303   
4035               0      1967             0    98004  47.6316 -122.202   
4067               0      1959             0    98055  47.4859 -122.205   
6174            1320      1977             0    98133  47.7565 -122.331   
9077            1070      1911             0    98105  47.6675 -122.316   
9452            2010      1968             0    98107  47.6732 -122.363   
10958           2020      1962             0    98004  47.6340 -122.221   
12885            600      1987             0    98105  47.6617 -122.289   
15070           1500      1979             0    98059  47.4786 -122.131   
15670           1140      1901             0    98112  47.6241 -122.305   
17235           1300      1959             0    98112  47.6310 -122.303   
18477           1620      1904             0    98112  47.6263 -122.314   
19302           1310      1961             0    98034  47.7111 -122.211   

       sqft_living15  sqft_lot15  
1660            1800        2580  
4035            2050        9100  
4067            1480        4942  
6174            1760        7624  
9077            1610        3000  
9452            1560        3737  
10958           3670       20500  
12885           2170        3750  
15070           1780       10457  
15670           1540        4800  
17235           4440        8640  
18477           4210        8325  
19302           1380        8172  

[13 rows x 21 columns]

In [12]:
df.loc[(df['bedrooms'] == 8) & (df['bathrooms'] == 2.75) ]

id       date      price  bedrooms  bathrooms  sqft_living  \
6174   1873400020 2014-07-03   340000.0         8       2.75         2790   
10958  1652500060 2014-07-11  1650000.0         8       2.75         4040   
15670  8141200080 2014-08-14   680000.0         8       2.75         2530   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
6174       6695     1.0           0     0  ...      7        1470   
10958     20666     1.0           0     0  ...      9        2020   
15670      4800     2.0           0     0  ...      7        1390   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
6174            1320      1977             0    98133  47.7565 -122.331   
10958           2020      1962             0    98004  47.6340 -122.221   
15670           1140      1901             0    98112  47.6241 -122.305   

       sqft_living15  sqft_lot15  
6174            1760        7624  
10958           3670       20500  
15670           1540        4800  

[3 rows x 21 columns]

For equal numbers of bedrooms and bathrooms, we have huge differences in the prices, for apartments of id "1873400020" and "8141200080", the price is doubled even though we have equal number of bedrooms, bathrooms, and very close number of sqaure feet of living space. This doubling of price may be due to "increment a level in floor" or "difference in latitudes and longtitudes hence being in a different area" as all of other factors doesn't differ much.
Also, the 10x increase of price of the "1652500060" house is influenced by square feet of living space and square feet of basement, also this house is in a different zipcode. All those would have increased the price of this house by this much. 

In [13]:
correlation = (df['bathrooms'] + df['bedrooms']).corr(df['price'])
print(correlation)


0.46631014225336537


In [14]:
correlation = (df['bathrooms'] + df['bedrooms'] + df['sqft_living']).corr(df['price'])
print(correlation)

0.7019418010114713


What is noticed that, an increase in the square feet of living space has a high influence, reassuring what was indicated by the correlation function that the sqft_living is of the highest influence on price. 

In [15]:
df[['price', 'bedrooms', 'bathrooms', 'sqft_living']].groupby(['sqft_living']).mean().round().reset_index()

sqft_living      price  bedrooms  bathrooms
0             290   142000.0       0.0        0.0
1             370   276000.0       1.0        1.0
2             380   245000.0       1.0        1.0
3             384   265000.0       0.0        1.0
4             390   236500.0       0.0        1.0
...           ...        ...       ...        ...
1033         9640  4670000.0       5.0        7.0
1034         9890  6890000.0       6.0        8.0
1035        10040  7060000.0       5.0        4.0
1036        12050  7700000.0       6.0        8.0
1037        13540  2280000.0       7.0        8.0

[1038 rows x 4 columns]

Through the above dataframe, as the square footage of living space increase, the price tends to increase as well. This shows a strong positive correlation between the square footage area and the price. Also, as the sqft living increase the number of bedrooms and bathrooms increase. Where it is typical that larger homes have more bedrooms and bathrooms.

In [16]:
df[['price', 'bedrooms', 'bathrooms', 'sqft_living', 'grade']].groupby(['grade']).mean().round().reset_index()

grade      price  bedrooms  bathrooms  sqft_living
0       1   142000.0       0.0        0.0        290.0
1       3   205667.0       1.0        0.0        597.0
2       4   214381.0       1.0        1.0        660.0
3       5   248524.0       2.0        1.0        983.0
4       6   301917.0       3.0        1.0       1192.0
5       7   402593.0       3.0        2.0       1689.0
6       8   542895.0       3.0        2.0       2185.0
7       9   773738.0       4.0        3.0       2868.0
8      10  1072347.0       4.0        3.0       3520.0
9      11  1497792.0       4.0        4.0       4395.0
10     12  2192500.0       4.0        4.0       5472.0
11     13  3710769.0       5.0        5.0       7483.0

There is a proportional relationship between the grade and the price. As the grade increases so does the average price per each grade. Hence a higher grades properties are associated with higher prices. Also, higher grades tend to have more sqft living, bathrooms and bedrooms, which would imply that higher grades have larger and more luxurious houses. This aligns with the result of the correlation function, where the "grade" attribute had the second highest influence on the price.

In [17]:
df[['price', 'bedrooms', 'bathrooms', 'sqft_living', 'grade', 'view']].groupby(['view']).mean().round(1).reset_index()

view      price  bedrooms  bathrooms  sqft_living  grade
0     0   496623.5       3.3        2.1       1997.8    7.6
1     1   812518.6       3.5        2.3       2569.0    8.1
2     2   792746.2       3.6        2.4       2655.3    8.3
3     3   972468.4       3.7        2.7       3018.6    8.7
4     4  1464362.6       3.6        2.8       3351.5    9.1

As the view level increase, so does the price. Meaning that properties of higher view levels (better views) often have higher prices. Hence, view levels play a significant role in the desirability of the appartments. People would pay more for appartments with better view, larger area, and better quality. The average price for an excellet view (view = 4) is multiplied by ten of that with no view (view = 0). This is usually the case because more luxurious houses often have better veiws along with better overall qualities. 

In [18]:
df[['price', 'bedrooms', 'bathrooms', 'sqft_living', 'grade', 'view', 'floors']].groupby(['floors']).mean().round(1).reset_index()

floors      price  bedrooms  bathrooms  sqft_living  grade  view
0     1.0   442219.6       3.2        1.8       1731.8    7.1   0.2
1     1.5   559044.9       3.5        1.8       1907.9    7.2   0.2
2     2.0   649051.5       3.6        2.6       2571.3    8.4   0.2
3     2.5  1061020.9       4.0        2.8       3145.4    8.9   0.5
4     3.0   582620.1       3.0        2.5       1788.3    8.2   0.3
5     3.5   933937.5       3.0        2.5       2518.0    8.4   0.5

The relationship between the number of floors and the price of the property is not always straightforward, where there are cases where the increase in the number of floors doesn't necessarily imply an increase in the price. This indicates that there are other attributes that do have an influence along with the number of floor, and this influence may be of better effect. The highest prices went for floors = 2.5 and 3.5, where the mean prices for those two floors nb is approximately double of the others, this may be due to the increase of sqtf_living accompanied with these two levels, and the increase of view for them as well (views = 0.5 higher than all other floors number)

We can deduce from this analysis that one factor may not necessarily affect the target at its own, however it may do so along with various factors, that together may give an increased influence on the target. That's why the analysis has to be on a larger dataset and including more attributes to obtain the relationships between them

In [19]:
df[['price', 'bedrooms', 'bathrooms', 'sqft_living', 'grade', 'view', 'waterfront']].groupby(['waterfront']).mean().round().reset_index()

waterfront      price  bedrooms  bathrooms  sqft_living  grade  view
0           0   531653.0       3.0        2.0       2072.0    8.0   0.0
1           1  1662524.0       3.0        3.0       3174.0    9.0   4.0

From this table we can conclude that the houses with waterfront often have higher prices than those without. Waterfront may be a desirable feature hence commanding a higher price. Eventhough there isn't much difference in the average number of bedrooms and bathrooms, sqft_living space, and grade (i.e. there isn't a huge relationship between these attributes and whether there is a waterfront or not). But they still may be impacting slightly. The view level increased significantly from 0 (with no waterfront) to 4 (with waterfront) explaining the huge increase in price as well. 

In [20]:
df[['price', 'waterfront', 'view', 'sqft_living']].groupby('waterfront')['view'].value_counts(normalize=True).mul(100).rename('percentage').reset_index()

waterfront  view  percentage
0           0     0   90.857809
1           0     2    4.452214
2           0     3    2.289044
3           0     1    1.543124
4           0     4    0.857809
5           1     4   82.822086
6           1     3   11.656442
7           1     2    4.907975
8           1     1    0.613497

The dataframe above shows that above 90% of the houses with no waterfront don't have a view, and a very small percentage of them (8%) have a view varying between (2-3-1-4). For houses with waterfront, however, the majority of them (83%) have an excellent view (view = 4), 11% have a verygood view, and the rest range between 2 and 1. None of the houses with waterfront is without a view. This indicates a very strong positive relationship between the waterfront and the view. This is expected as these two features are more of a luxurious features that require higher prices

In [21]:
df[['price', 'sqft_living', 'sqft_lot', 'sqft_basement']].groupby(['sqft_basement']).mean().round(1).reset_index()


sqft_basement      price  sqft_living  sqft_lot
0                0   486945.4       1928.9   16284.2
1               10   191900.0       1000.0    1092.0
2               20   485000.0       1430.0     923.0
3               40   400425.0       1245.0    1100.5
4               50   477358.0       1589.5    3306.5
..             ...        ...          ...       ...
301           3260  1900000.0       6510.0   16471.0
302           3480  7700000.0      12050.0   27600.0
303           3500  3200000.0       7000.0   28206.0
304           4130  2280000.0      13540.0  307752.0
305           4820  4670000.0       9640.0   13068.0

[306 rows x 4 columns]

There isn't a clear linear relationship between the sqft basement and the prices of the houses. Where houses with different basement area had various prices with no precise trend. Thus, the basement alone doesnt have a huge effect. However it is clear that basements with an big areas are more expensive. 

In [23]:
df[['price', 'sqft_living', 'grade', 'view', 'yr_built', 'yr_renovated']].groupby(['yr_built', 'yr_renovated']).mean().round(1).reset_index()


yr_built  yr_renovated     price  sqft_living  grade  view
0        1900             0  575245.4       1623.6    6.8   0.3
1        1900          1955  550000.0       2450.0    7.0   0.0
2        1900          1965  214000.0       1020.0    6.0   0.0
3        1900          1973  450000.0       1540.0    7.0   0.0
4        1900          1980  452000.0       2270.0    6.0   0.0
..        ...           ...       ...          ...    ...   ...
857      2011             0  544648.4       2313.7    8.2   0.1
858      2012             0  527437.0       2378.7    8.2   0.1
859      2013             0  678599.6       2636.6    8.5   0.2
860      2014             0  683792.7       2615.4    8.6   0.1
861      2015             0  759970.9       2596.3    8.9   0.1

[862 rows x 6 columns]

In [29]:
df.loc[(df['yr_built'] <2000 ) & (df['yr_renovated'] == 0), ['yr_built', 'yr_renovated', 'price', 'sqft_living'] ].sort_values(by='price', ascending=False)

yr_built  yr_renovated      price  sqft_living
1315       1991             0  5300000.0         7390
1164       1999             0  5110000.0         8010
2085       1980             0  3850000.0         5770
7035       1978             0  3800000.0         7050
19017      1997             0  3800000.0         5510
...         ...           ...        ...          ...
8274       1954             0    82000.0          860
16198      1943             0    81000.0          730
465        1912             0    80000.0          430
15293      1942             0    78000.0          780
1149       1966             0    75000.0          670

[15941 rows x 4 columns]

In [30]:
df.loc[(df['yr_built'] <2000 ) & (df['yr_renovated'] > 0), ['yr_built', 'yr_renovated', 'price', 'sqft_living']].sort_values(by='price', ascending=False)

yr_built  yr_renovated      price  sqft_living
7252       1910          1987  7700000.0        12050
3914       1940          2001  7060000.0        10040
8092       1983          2009  4670000.0         9640
2864       1941          2002  3600000.0         5020
10446      1939          2010  3400000.0         4260
...         ...           ...        ...          ...
19417      1937          1946   152275.0         1020
12984      1969          2009   146000.0          900
15540      1943          1989   135000.0          600
2589       1922          2003   134000.0          980
20962      1968          2007   110000.0          828

[913 rows x 4 columns]

Through the above dataframes, we can get many insights. 
First of all, for houses that were built before 2000 and weren't renovated, the highest prices were for the houses was were built near 2000, and the early constructed houses had much less prices. Meaning that for those houses without renovation, the newer ones had higher prices. 
Second, concerning the houses that were built before 2000 but were renovated, and despite the year they were renovated, they did have much higher prices (reaching 7 million dollars), also they had larger area. The renovated houses tend to have larger prices than those that weren't renovated. Hence the act of renovation has a positive influence on the price.
It appears that the year the house was built alone doesn't have a strong relationship with the price, however when considering the year the house was built along with the year it was renovated, the correlation with the price becomes more significant.   

In [33]:
df.loc[(df['yr_built'] >2000 ) & (df['yr_renovated'] > 0), ['yr_built', 'yr_renovated', 'price', 'sqft_living']].sort_values(by='price', ascending=False)

yr_built  yr_renovated     price  sqft_living
18185      2003          2009  825000.0         3620

Of the houses that were built after 2000 only one was renovated, which is logical because they have been built lately and don't need much modifications. (As the data was collected on 2014, hence not much of a time difference.)

In [35]:
df.loc[(df['yr_built'] >2000 ) , ['yr_built', 'yr_renovated', 'price', 'sqft_living']].sort_values(by='price', ascending=False)

yr_built  yr_renovated      price  sqft_living
9254       2001             0  6890000.0         9890
4411       2001             0  5570000.0         9200
1448       2009             0  5350000.0         8000
2626       2004             0  4500000.0         6640
8638       2001             0  4490000.0         6430
...         ...           ...        ...          ...
20807      2014             0   182568.0         1500
19975      2014             0   175003.0         1390
20612      2004             0   163800.0         1000
20403      2012             0   160797.0         1270
19650      2013             0   155000.0         1300

[4541 rows x 4 columns]

The houses that were built near 2000 had higher prices than those that were built after 2010, this may be due to many reasons; design, styling of the houses, the areas the houses are in.. We have to check for the other attributes along with this to include any trend. 

Through this thorough analysis i did, i realized that multiple factors contribute to house pricing much more than single factors alone. Eventhough the square footage of living space does have a huge impact, yet this impact is also influenced by other factors. Most of the attributes alone don't have the highest influence but when gathered the influence increased. 
To do such an analysis many cases have to be taken on all of the attributes, and they have to be crossed with many other attributes in order to find trends and patterns on the price.